src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs

```diff
@@ -430,6 +434,27 @@ private AntiforgeryTokenSet Serialize(IAntiforgeryFeature antiforgeryFeature)
                 _options.HeaderName);
         }
 
+        private bool TryDeserializeTokens(
+            HttpContext httpContext,
+            AntiforgeryTokenSet antiforgeryTokenSet,
+            out AntiforgeryToken cookieToken,
+            out AntiforgeryToken requestToken)
+        {
+            try
+            {
+                DeserializeTokens(httpContext, antiforgeryTokenSet, out cookieToken, out requestToken);
+                return true;
+            }
+            catch (Exception ex)
```

Comment:
I have the same feedback I have about the previous code. This is a try/catch that surrounds rather **a lot** of code. The try/catch needs to move down a layer into the serializer **or** this needs to catch specific exception type.

This has the same risks as the input formatter thing 😆

Reply:
Yeah, I pondered about this and used `Exception` as this is a `Try..` method, but as you mentioned (and I agree) that catching this root exception type is never good.